dojo/unittests/tools/test_sonarqube_importer.py

```diff
@@ -24,6 +24,74 @@ def dummy_rule(self, *args, **kwargs):
+        return data
+
+
+def dummy_many_hotspots(self, *args, **kwargs):
+    with open('dojo/unittests/scans/sonarqube_api/hotspots/many_vulns.json') as json_file:
+        data = json.load(json_file)
+        return data
+
+
+def dummy_hotspot_rule(self, *args, **kwargs):
+    with open('dojo/unittests/scans/sonarqube_api/hotspots/rule.json') as json_file:
+        data = json.load(json_file)
+        return data
+
+
+def empty_list(self, *args, **kwargs):
+    return list()
+
+
+def verify_issues_fields_match_findings_fields(self, parser, finding, issue, *args, **kwargs):
+    rule = dummy_rule(self)
+    self.assertIsNotNone(issue)
+    self.assertEqual(str(finding.title), str(issue['message']))
+    self.assertEqual(str(finding.cwe), str(parser.clean_cwe(rule['htmlDesc'])))
+    self.assertEqual(str(finding.description), str(parser.clean_rule_description_html(rule['htmlDesc'])))
+    self.assertEqual(str(finding.severity), str(parser.convert_sonar_severity(issue['severity'])))
+    self.assertEqual(str(finding.references), str(str(parser.get_references(rule['htmlDesc']))))
+    self.assertEqual(str(finding.file_path), str(issue['component']))
+    self.assertEqual(str(finding.line), str(issue['line']))
+    self.assertEqual(str(finding.verified), str(parser.is_confirmed(issue['status'])))
+    self.assertEqual(finding.false_p, False)
+    self.assertEqual(finding.duplicate, False)
+    self.assertEqual(finding.out_of_scope, False)
+    self.assertEqual(finding.mitigated, None)
+    self.assertEqual(str(finding.mitigation), "No mitigation provided")
+    self.assertEqual(str(finding.impact), "No impact provided")
+    self.assertEqual(finding.static_finding, True)
+    self.assertEqual(str(finding.sonarqube_issue), str(issue['key']))
+
+
+def verify_hotspots_fields_match_findings_fields(self, parser, finding, hotspot, *args, **kwargs):
+    rule = dummy_hotspot_rule(self)
+    self.assertIsNotNone(hotspot)
+    self.assertEqual(str(finding.title), str(hotspot['message']))
+    self.assertEqual(str(finding.cwe), str(parser.clean_cwe(rule['riskDescription'])))
+    self.assertEqual(str(finding.description), str(parser.clean_rule_description_html(rule['vulnerabilityDescription'])))
+    self.assertEqual(str(finding.severity), str(parser.convert_sonar_review_priority(hotspot['vulnerabilityProbability'])))
+    self.assertEqual(str(finding.references), str(str(parser.get_references(rule['riskDescription']))))
+    self.assertEqual(str(finding.file_path), str(hotspot['component']))
+    self.assertEqual(str(finding.line), str(hotspot['line']))
+    self.assertEqual(finding.active, True)
+    self.assertEqual(str(finding.verified), str(parser.is_confirmed(hotspot['status'])))
+    self.assertEqual(finding.false_p, False)
+    self.assertEqual(finding.duplicate, False)
+    self.assertEqual(finding.out_of_scope, False)
+    self.assertEqual(finding.static_finding, True)
+    self.assertEqual(str(finding.sonarqube_issue), str(hotspot['key']))
```

Comment:
Something is wrong here.
We are testing exactly how the code of the parser analyze a dict.
It's too low level.
You need to specialize one unit test with the data of a specific report.

Reply:
@damiencarol So, just to clarify the comment above:

1. Should I leave the tests to check the number of findings and create a new test to check the report fields?
2. Let's take for example the title in a finding, should I test it against the string itself and not from the loaded issue or hotspot?

Reply:
You can just test the fields of the first findings that come from hotpots data in the test below but consider the `get_findings` method as a black-box.
If you make assumptions on which field is read and link it in the unit test your are just re-writing the code of the parser in the unit tests. Which means it's duplicate code, which means 2x the work if the data structure of SQ is changing or if we modify the implementation of the parser.
Just add few more checks on the result of `get_findings` method in the news tests you wrote.

Reply:
Done, if I understood it correctly.